jupyterhub/templates/NOTES.txt

```diff
@@ -20,6 +20,17 @@ Note that this is still an alpha release! If you have questions, feel free to
   3. File issues at https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues
 
 
+{{- if or .Values.proxy.secretToken .Values.hub.config.JupyterHub.proxy_auth_token }}{{ println }}
+NOTE: proxy.secretToken or hub.config.JupyterHub.proxy_auth_token was explicitly set, this is no longer needed.
```

Comment:
I don't think we need to discourage setting these values. While it is no longer necessary, that doesn't mean it is a bad practice (values are stable across full helm delete & redeploy, for instance).

Reply:
I see a value in it this encouragement because this is the only secret that we need outside of the hub container afaik which increases the complexity of solving #2009 for example.

If people remove it from their configs, the examples they copy paste in forums etc will not propegate the practice either.

Would you consider it a better practice to let this be managed explicitly by a user? I figure this is ephemeral configuration that doesn't require any state change other than a restart of the helm chart managed pods, and it better to not set it explicitly as a user.

Reply:
I don't think it's better to not set it, or necessarily better to set it. I think it's a big improvment to not *have* to set it, but that's not the same as the user shouldn't be *able* to manage it in configuration, or even discouraged.

I think for the most part, behavior should be completely unchanged *except* in the event of the value being unset, in which case it has this 'persistent' generated default value. Certainly setting it manually should have precedence.

Reply:
I'll remove this note as part of this PR.

Note that explicitly set values have precedence.

Reply:
Resolved by b8c60ef